tools/explorer-thumbnailer/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+use com::registration::{RegistryKeyInfo, dll_register_server, dll_unregister_server};
+use thumbnail_provider::{CLSID_THUMBNAIL_PROVIDER_CLASS, ThumbnailProvider};
+
+mod error;
+mod utils;
+mod interfaces;
+mod thumbnail_provider;
+
+// we replace the com::registration::inproc_dll_module macro in order to be able
+// to modify DllRegisterServer and DllUnregisterServer functions
```

Comment:
COM-rs maintainer here 👋. I wonder if we should provide hooks so that you don't have to copy/paste these definitions over.

Reply:
Yes, that would be great! Also `impl Display for GUID` please, so we can use it in registry key paths like

https://github.com/gentoo90/resvg/blob/explorer-thumbnailer/tools/explorer-thumbnailer/src/lib.rs#L67